src/com/amazon/corretto/crypto/provider/Utils.java

```diff
@@ -524,5 +525,14 @@ static int getJavaVersion() {
         }
         return JAVA_VERSION;
     }
+
+    static boolean getCacheSelfTestResultsProperty() {
+        final String relyOnCacheSelfTestResultsStr = Loader.getProperty(CACHE_SELF_TEST_RESULTS, "true").toLowerCase();
+        if (!relyOnCacheSelfTestResultsStr.equals("true") && !relyOnCacheSelfTestResultsStr.equals("false")) {
+            System.err.format("Valid values for %s are false and true, with true as default", CACHE_SELF_TEST_RESULTS);
+            return true;
```

Comment:
why not just rely on parseBoolean for error handling? it seems like we'd want to fail loudly if an invalid property value was specified so the user knows about the issue

Reply:
I just followed the other system variable reading: in case of failure, default to a safe value, which in this case is the existing behavior.

Reply:
hm, that's a good point. reading over the class again, i agree that [property parsing tends](https://github.com/corretto/amazon-corretto-crypto-provider/blob/develop/src/com/amazon/corretto/crypto/provider/Utils.java#L482) towards "fall back to [safe default](https://github.com/corretto/amazon-corretto-crypto-provider/blob/develop/src/com/amazon/corretto/crypto/provider/Utils.java#L523)" in error cases.